lib/backends/smartos/tasks/machine_migrate.js

```diff
@@ -803,6 +803,76 @@ function setAutoboot(callback) {
 }
 
 
+function setIndestructibleZoneroot(callback) {
+    var log = this.log;
+    var payload = this.req.params;
+
+    assert.object(payload, 'payload');
+    assert.uuid(payload.vm_uuid, 'payload.vm_uuid');
+    assert.string(payload.value, 'payload.value');
+
+    var vmUuid = payload.vm_uuid;
+    var value = payload.value;
+
+    // Update using vmadm.
+    var cmd = '/usr/sbin/vmadm';
+    var args = [
+        'update',
+        vmUuid,
+        'indestructible_zoneroot=' + value
+    ];
+
+    var buf;
+
+    log.debug({cmd: cmd, args: args}, 'setIndestructibleZoneroot');
+
+    try {
+        buf = child_process.execFileSync(cmd, args, gExecFileDefaults);
+    } catch (ex) {
+        log.warn({cmd: cmd, args: args}, 'Could not run vmadm update:', ex);
+        this.fatal(String(ex.stderr || buf));
+        return;
+    }
+
+    this.finish();
+}
+
+
+function setIndestructibleDelegated(callback) {
+    var log = this.log;
+    var payload = this.req.params;
+
+    assert.object(payload, 'payload');
+    assert.uuid(payload.vm_uuid, 'payload.vm_uuid');
+    assert.string(payload.value, 'payload.value');
+
+    var vmUuid = payload.vm_uuid;
+    var value = payload.value;
+
+    // Update using vmadm.
+    var cmd = '/usr/sbin/vmadm';
+    var args = [
+        'update',
+        vmUuid,
+        'indestructible_delegated=' + value
+    ];
+
+    var buf;
+
+    log.debug({cmd: cmd, args: args}, 'setIndestructibleDelegated');
+
+    try {
+        buf = child_process.execFileSync(cmd, args, gExecFileDefaults);
+    } catch (ex) {
+        log.warn({cmd: cmd, args: args}, 'Could not run vmadm update:', ex);
+        this.fatal(String(ex.stderr || buf));
+        return;
+    }
+
+    this.finish();
+}
+
+
 function start(callback) {
```

Comment:
The `start` function does not need a `callback`.  If you check recent tasks like `recovery_config` or `uninstall_agents` you'll see that those aren't using it at all. My bet is that it's just legacy convention from a past when it was needed.